beautifultable/beautifultable.py

```diff
@@ -1213,3 +1213,53 @@ def from_csv(self, file_name, header=True, **kwargs):
             for row in csv_reader:
                 self.rows.append(row)
             return self
+
+    def to_df(self):
+        """Export table to dataframe.
+
+        Parameters
+        ----------
+        table : BeautifulTable
+
+        """
+        try:
+            pd = __import__("pandas")
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError("Please Install pandas to use this API") from e
+
+        # If there are column headers then it will act as a column of datafarme
+        headers = list(self.columns.header)
+        if headers.count(None) == len(headers):
+            headers = None
+
+        # If there are row headers then it will act as an Index
+        index = list(self.rows.header)
+        if index.count(None) == len(index):
+            index = None
+
+        return pd.DataFrame(
+            [list(row) for row in self.rows], columns=headers, index=index
+        )
+
+    def from_df(self, df):
+        """Import table from dataframe.
+
+        Parameters
+        ----------
+        df : pandas.Dataframe
+        """
+        data = df.to_dict()
+
+        # Dataframe columns will act as a column headers
+        headers = list(data.keys())
+
+        # Index of dataframe will act as a row headers
+        row_header = list(df.index)
+
+        for header in headers:
+            self.columns.append(
+                [data[header][indx] for indx in row_header], header=str(header)
+            )
+        if list(range(len(row_header))) != row_header:
+            self.rows.header = row_header
```

Comment:
in case of reading from a dataframe, let's just use the indexes as row headers as is. so, by default, let that be [0,1,2,...]. If the user wants to hide them, they'll add an explicit call `table.rows.headers = None` after importing the dataframe.